### PR TITLE
[BISERVER-12300] - Metadata on files/folders erased during export/import

### PR DESCRIPTION
**Pull request by bcostahitachivantara**
@pentaho-lmartins 
@pentaho/tatooine

### CODE REVIEW

**Comment by buildguy** [comment]
### :x: Build failed in 6h 9s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.platform.plugin.services.importer.RepositoryFileImportBundleTest.testGettersAndSetters</b> (click to expand)</summary><p>

```

Expected: bean with valid setter and getter methods for all properties
     but: bean of type "org.pentaho.platform.plugin.services.importer.RepositoryFileImportBundle" had an invalid getter/setter for the property "charset"
```

</p></details>

  
Tests run: **2542**, Failures: **1**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 6h 18s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.platform.plugin.services.importer.RepositoryFileImportBundleTest.testGettersAndSetters</b> (click to expand)</summary><p>

```

Expected: bean with valid setter and getter methods for all properties
     but: bean of type "org.pentaho.platform.plugin.services.importer.RepositoryFileImportBundle" had an invalid getter/setter for the property "charset"
```

</p></details>

  
Tests run: **2542**, Failures: **1**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 29m 27s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: extensions/src/main/java/org/pentaho/platform/plugin/services/importexport/exportManifest/ExportManifestEntity.java [hunk 1]
```diff
@@ -77,9 +91,34 @@ protected ExportManifestEntity( File file, String userId, String projectId, Bool
...
     createEntityAcl( userId );
     rawExportManifestProperty.setEntityMetaData( entityMetaData );
+    rawExportManifestProperty.setEntityExtraMetaData( entityExtraMetaData );
     rawExportManifestProperty.setEntityAcl( entityAcl );
   }
 
+  private void createEntityExtraMetaData( RepositoryFile repositoryFile ) {
+    entityExtraMetaData = new EntityExtraMetaData();
+
+    if ( repositoryFile instanceof RepositoryFileProxy ) {
```
Comment (LeonardoCoelho71950) [inline extensions/src/main/java/org/pentaho/platform/plugin/services/importexport/exportManifest/ExportManifestEntity.java]: I understand this solution.
However, I not sure if that has no impact. Namely, if we decide to extends RepositoryFile in new ways...

**Comment by buildguy** [comment]
### :x: Build failed in 6h 23s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 31m 34s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 6h 9s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/6/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: repository/src/main/java/org/pentaho/platform/repository2/unified/jcr/RepositoryFileProxy.java [hunk 2]
```diff
@@ -223,6 +222,24 @@ public void execute( Session session ) {
...
 
+  public void setMetadata( Map<String, Serializable> metadata ) throws RepositoryException {
+
+      this.executeOperation( new SessionOperation() {
+        @Override
+        public void execute( Session session ) {
+          try {
+            JcrRepositoryFileUtils.setFileMetadata( session, getId(), metadata );
+          } catch ( InvalidItemStateException | ItemNotFoundException | PathNotFoundException e ) {
+            getLogger().warn( "InvalidItemStateException in setMetadata. Probable cause: File does not exist anymore" );
```
[the remark this answers is not in the document]
Comment (bcostahitachivantara) [reply repository/src/main/java/org/pentaho/platform/repository2/unified/jcr/RepositoryFileProxy.java]: In this file we do not have any string localized. I can do it in a new PR with just improvements.

File: repository/src/main/java/org/pentaho/platform/repository2/unified/jcr/RepositoryFileProxy.java [hunk 3]
```diff
@@ -223,6 +222,24 @@ public void execute( Session session ) {
...
+          } catch ( RepositoryException e ) {
+            getLogger().error( "RepositoryException was found: ", e );
```
[the remark this answers is not in the document]
Comment (bcostahitachivantara) [reply repository/src/main/java/org/pentaho/platform/repository2/unified/jcr/RepositoryFileProxy.java]: In this file we do not have any string localized. I can do it in a new PR with just improvements.

File: extensions/src/main/java/org/pentaho/platform/plugin/services/importer/PDIImportFileHandler.java [hunk 4]
```diff
@@ -59,7 +59,7 @@ protected RepositoryFile updateFile( final RepositoryFileImportBundle bundle, fi
       getRepository().deleteFile( file.getId(), true, null );
 
       RepositoryFileAcl newFileAcl = bundle.getAcl();
-      bundle.setAcl( originFileAcl );
```
[the remark this answers is not in the document]
Comment (bcostahitachivantara) [reply extensions/src/main/java/org/pentaho/platform/plugin/services/importer/PDIImportFileHandler.java]: Good catch!

File: extensions/src/main/java/org/pentaho/platform/plugin/services/importer/RepositoryFileImportBundle.java [hunk 5]
```diff
@@ -287,8 +304,8 @@ public Builder schedulable( Boolean schedulable ) {
       return this;
     }
 
-    public Builder charSet( String charSet ) {
-      bundle.setCharSet( charSet );
+    public Builder charSet( String charset ) {
```
[the remark this answers is not in the document]
Comment (bcostahitachivantara) [reply extensions/src/main/java/org/pentaho/platform/plugin/services/importer/RepositoryFileImportBundle.java]: I agree with your comment about consistency, so I will rename the changes to "charSet".

File: extensions/src/main/java/org/pentaho/platform/plugin/services/importer/RepositoryFileImportFileHandler.java [hunk 6]
```diff
@@ -317,6 +331,38 @@ private RepositoryFileAcl getDefaultAcl( RepositoryFile repositoryFile ) {
...
+   * @param newFile
+   *          Whether the file is being newly created or was pre-existing
+   * @param repositoryFileExtraMetaData
+   *          The effect extraMetaData as defined in the manifest
+   * @param repositoryFile
+   *          The <code>RepositoryFile</code> of the target file
+   */
+  private void updateExtraMetaData( boolean newFile, RepositoryFile repositoryFile, RepositoryFileExtraMetaData repositoryFileExtraMetaData ) {
+    if ( repositoryFileExtraMetaData != null && !repositoryFileExtraMetaData.getExtraMetaData().isEmpty()) {
+      getLogger().debug( "Apply extra meta data to file that " + ( newFile ? "is new" : "already exists" ) );
```
[the remark this answers is not in the document]
Comment (bcostahitachivantara) [reply extensions/src/main/java/org/pentaho/platform/plugin/services/importer/RepositoryFileImportFileHandler.java]: This message is just to debug. Using the same standard for this class debug and trace log are not localized.

File: extensions/src/main/java/org/pentaho/platform/plugin/services/importer/RepositoryFileImportFileHandler.java [hunk 7]
```diff
@@ -317,6 +331,38 @@ private RepositoryFileAcl getDefaultAcl( RepositoryFile repositoryFile ) {
...
+   *          The RepositoryImportBundle (which contains the effective manifest extraMetaData)
+   * @param repositoryFile
+   *          The <code>RepositoryFile</code> of the target file
+   */
+  private void updateExtraMetaDataFromBundle( boolean newFile, RepositoryFileImportBundle bundle, RepositoryFile repositoryFile ) {
+    updateExtraMetaData( newFile, repositoryFile, bundle.getExtraMetaData() );
+  }
+
+  /**
+   * Create a formal <code>RepositoryFileAcl</code> object for import.
```
[the remark this answers is not in the document]
Comment (bcostahitachivantara) [reply extensions/src/main/java/org/pentaho/platform/plugin/services/importer/RepositoryFileImportFileHandler.java]: Yes, you rigth

File: extensions/src/main/java/org/pentaho/platform/plugin/services/importexport/exportManifest/ExportManifestEntity.java (hunk 1 above)
Comment (bcostahitachivantara) [reply extensions/src/main/java/org/pentaho/platform/plugin/services/importexport/exportManifest/ExportManifestEntity.java]: I understand with your concern, but this is a solution it less impact. 
As an alternative solution could be change parent class but I am not sure if metadata have any meaning in RepositoryFile itself.

I am open for suggestion :)

**Comment by buildguy** [comment]
### :x: Build failed in 4h 47m 12s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/7/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 43m 53s
Build command:
```console
mvn clean verify -X -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/8/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 5s
Build command:
```console
mvn clean verify -B -e -Daudit -amd \
-Dsonar-maven-plugin.version=3.7.0.1746
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Comment by tmcsantos** [comment]
This PR is hanging on a sonar rule validation. We are investigating

**Comment by buildguy** [comment]
### :x: Build failed in 35m 47s
Build command:
```console
mvn clean verify -B -e -Daudit -amd \
-Dsonar-maven-plugin.version=3.7.0.1746 -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/10/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 35m 48s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/11/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 30m 4s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/12/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 6s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 53s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 12m
Build command:
```console
mvn clean verify -X -B -e -Daudit -DskipTests -amd -pl \
api,extensions,repository
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 33m 54s
Build command:
```console
mvn clean verify -X -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/16/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 2h 15m 25s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/17/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 29m 13s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/18/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by pentaho-lmartins** [comment]
@bmorrise @pentaho/r2d2 can you please review this. Thank you!

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 30m 5s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/19/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by bmorrise** [comment]
SonarQube is reporting a number of issues. Are those being addressed?

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 29m 59s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/20/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 6h 17s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/21/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 10m 57s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **624**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/22/testReport)

 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 15m 51s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **1065**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/23/testReport)

 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 15m 45s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **1065**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/25/testReport)

 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 6h 10s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/26/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 29m 45s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/27/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 6h 21s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/28/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 29m 30s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/29/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 6h 11s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/30/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 6h 27s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/31/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 6h 13s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/32/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 6h 23s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/33/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 6h 22s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/34/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 6h 15s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/35/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 11m 16s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl repository
```
---
:ok_hand: All tests passed!
 
Tests run: **388**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/2/testReport)

 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 8m 21s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl repository
```
---
:ok_hand: All tests passed!
 
Tests run: **388**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/3/testReport)

 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 7m 31s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl repository
```
---
:ok_hand: All tests passed!
 
Tests run: **388**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/4/testReport)

 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 1h 6m 23s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 1h 39m 58s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/6/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 19m 43s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 54m 44s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/8/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by bcostahitachivantara** [comment]
Those builds failed are related with:
https://jira.pentaho.com/browse/BACKLOG-33232

And exists a sonar issue too that was already reported by João França

**Comment by buildguy** [comment]
### :x: Build failed in 6h 16s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2548**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/9/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 1m 8s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 51m 17s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2551**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/11/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 32m 32s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2551**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/12/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by bcostahitachivantara** [comment]
Hello @bmorrise, this PR finally had a successful build, could you review this?

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 32m 47s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2571**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/13/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 32m 43s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,extensions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **2542**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman2.pentaho.net/job/projects/job/platform/job/PR-4585/14/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**